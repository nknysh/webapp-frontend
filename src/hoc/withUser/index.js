export { default } from './withUser';
export { propTypes } from './withUser.props';
