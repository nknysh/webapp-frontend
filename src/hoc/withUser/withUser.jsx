import React from 'react';
import { propOr, compose } from 'ramda';

import connect from './withUser.state';

const withUser = WrappedComponent => {
  const ConnectedWrappedComponent = compose(connect)(WrappedComponent);

  const wrappedDisplayName = propOr(propOr('Component', 'name', WrappedComponent), 'displayName', WrappedComponent);

  const WithUser = props => <ConnectedWrappedComponent {...props} />;
  WithUser.displayName = `withUser(${wrappedDisplayName})`;

  return WithUser;
};

export default withUser;
