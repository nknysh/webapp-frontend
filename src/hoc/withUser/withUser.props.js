import PropTypes from 'prop-types';

export const propTypes = {
  user: PropTypes.object,
  isSr: PropTypes.bool,
};
