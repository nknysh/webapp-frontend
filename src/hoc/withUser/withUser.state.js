import { connect } from 'react-redux';

import { isSR, isRL, getCurrentUser } from 'store/modules/auth';

const mapStateToProps = state => ({
  isSr: isSR(state),
  isRl: isRL(state),
  user: getCurrentUser(state),
});

export default connect(mapStateToProps);
