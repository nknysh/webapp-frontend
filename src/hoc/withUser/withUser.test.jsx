import React from 'react';
import { compose } from 'ramda';

import withUser from './withUser';

const MockComponent = () => <div />;

describe('withUser HOC', () => {
  describe('render', () => {
    it('connected component', () => {
      const Component = compose(withUser)(MockComponent);
      const mountedComponent = shallow(<Component />);
      expect(mountedComponent).toMatchSnapshot();
    });
  });
});
