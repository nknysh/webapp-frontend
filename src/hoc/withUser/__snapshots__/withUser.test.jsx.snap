// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUser HOC render connected component 1`] = `<Connect(MockComponent) />`;
